src/generic_ui/scripts/ui.ts

```diff
@@ -538,14 +542,22 @@ module UI {
       */
     public startGettingInUiAndConfig =
         (instanceId :string, endpoint :Net.Endpoint) => {
-      this.instanceGettingAccessFrom_ = instanceId;
+      if (instanceId) {
+        this.instanceGettingAccessFrom_ = instanceId;
+        this.mapInstanceIdToUser_[instanceId].isSharingWithMe = true;
+      }
 
       this.startGettingInUi();
 
       this.updateGettingStatusBar_();
 
-      this.mapInstanceIdToUser_[instanceId].isSharingWithMe = true;
+      if (this.proxySet_) {
+        // this handles the case where the user starts proxying again before
+        // confirming the disconnect
+        this.stopGettingInUiAndConfig(false);
```

Comment:
Is there any chance of a race condition here?  The documentation for chrome (https://developer.chrome.com/extensions/proxy) makes it look like setting/clearing the proxy is async.  Not sure how that works in practice, but we wouldn't want to get into a situation where the ```this.browserApi.startUsingProxy(endpoint);``` call below completes only be cleared as a result of an async call to undo the proxy.

Maybe in this case ```this.stopGettingInUiAndConfig``` should skip calling ```this.browserApi.stopUsingProxy``` since we are just going to be changing the proxy to something else?